src/main/java/uk/gov/companieshouse/api/strikeoffobjections/authorization/UserAuthorizationInterceptor.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.companieshouse.api.strikeoffobjections.authorization;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.Objection;
+import uk.gov.companieshouse.api.strikeoffobjections.service.IObjectionService;
+import uk.gov.companieshouse.api.strikeoffobjections.service.impl.ERICHeaderFields;
+import uk.gov.companieshouse.api.strikeoffobjections.service.impl.ERICHeaderParser;
+import uk.gov.companieshouse.service.ServiceException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UserAuthorizationInterceptor implements HandlerInterceptor {
+
+    private final ApiLogger apiLogger;
+    private final  IObjectionService objectionService;
+    private final ERICHeaderParser ericHeaderParser;
+
+    public UserAuthorizationInterceptor(ApiLogger apiLogger, IObjectionService objectionService, ERICHeaderParser ericHeaderParser) {
+        this.apiLogger = apiLogger;
+        this.objectionService = objectionService;
+        this.ericHeaderParser = ericHeaderParser;
+
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+
+        final String requestId = request.getHeader(ERICHeaderFields.ERIC_REQUEST_ID);
+        apiLogger.debugContext(requestId, "Checking current user is authorised for objection id");
+        final String user = request.getHeader(ERICHeaderFields.ERIC_AUTHORISED_USER);
+        final String objectionId = getObjectionId(request.getRequestURI());
+
+        final Objection objection = objectionService.getObjection(requestId, objectionId);
+
+        final String createdByUserEmail = objection.getCreatedBy().getEmail();
+        final String requestUserEmail = ericHeaderParser.getEmailAddress(user);
+
+        if(createdByUserEmail.equals(requestUserEmail)) {
+            apiLogger.debugContext(requestId, "User is authorised for objection id");
+            return true;
+        }
+
+        apiLogger.debugContext(requestId, "User not authorised for objection id");
+        response.setStatus(HttpStatus.UNAUTHORIZED.value());
+        return false;
+    }
+
+    private String getObjectionId(String requestURI) throws ServiceException{
```

Comment:
I'm a bit uneasy about this logic (living here, I suppose). The URIs are all specified in the controller and I suspect they could change there and the unit test for this class would still pass. Perhaps an integration test would mitigate the risk...

Alternatively, the controller methods benefit from the @PathVariable annotation which automatically extracts the objection id. Can we somehow make use of that approach in this interceptor too?